pyradiosky/skymodel.py

```diff
@@ -2634,11 +2634,18 @@ def read_fhd_catalog(
                 beam_amp[2, src] = catalog["beam"][src]["XY"][0]
                 beam_amp[3, src] = catalog["beam"][src]["YX"][0]
 
+        if len(np.unique(ids)) != len(ids):
+            warnings.warn("WARNING: Source IDs are not unique. Defining unique IDs.")
+            unique_ids, counts = np.unique(ids, return_counts=True)
+            for repeat_id in unique_ids[np.where(counts > 1)[0]]:
+                fix_id_inds = np.where(ids == repeat_id)[0]
+                for append_val, id_ind in enumerate(fix_id_inds):
+                    ids[id_ind] = "{}{}".format(ids[id_ind], append_val + 1)
+
```

Comment:
Are these ids usually numerical for FHD catalogs? Or can they contain strings like source names? It might be better to put an underscore or some other marker between the original and the added number. I'm imagining names that have "telephone numbers" like in GLEAM and just appending a digit to the end might make it more confusing.

Actually, if they are just numbers, then adding a digit without an underscore might make them clash with another id in the list as well.

Reply:
I just fixed this. If the source IDs repeat, then another number is appended with a dash. For example, source IDs ["1", "1"] would become ["1-1", "1-2"]. Components have a number appended with an underscore.